### ISSUE TITLE
paddle2.4版本调用grad返回None

### ISSUE THREAD

**Issue by zhangyimi** [PaddlePaddle/Paddle#48257]
### bug描述 Describe the Bug

升级paddle 2.4.0rc或2.4.0版本后，已有功能出错，用户提了相关issue。https://github.com/PaddlePaddle/PaddleNLP/issues/3686

分析后发现是下列代码中`gradients = target_feature_map[0].grad `中`gradients`为None导致。（target_feature_map[0].stop_gradients=False）

这段代码在paddle2.4之前的版本功能正常，而2.4rc和2.4版本稳定复现，辛苦paddle同学看下是2.4存在不兼容升级还是存在bug。

```python
def  predict_fn (inputs, label, left=None, right=None, steps=None, paddle_model=None, embedding_name=None):

    if paddle_model is None:
        paddle_model = self.paddle_model
    target_feature_map = []

    def hook(layer, input, output):
        if steps is not None:
            noise_array = np.arange(left, right) / steps
            output_shape = output.shape
            assert right - left == output_shape[0]
            noise_tensor = paddle.to_tensor(noise_array, dtype=output.dtype)
            noise_tensor = noise_tensor.unsqueeze(axis=list(range(1, len(output_shape))))
            output = noise_tensor * output
        target_feature_map.append(output)
        return output

    hooks = []
    for name, v in paddle_model.named_sublayers():
        if embedding_name in name:
            h = v.register_forward_post_hook(hook)
            hooks.append(h)

    if isinstance(inputs, (tuple, list)):
        logits = paddle_model(*inputs)  # get logits, [bs, num_c]
    else:
        logits = paddle_model(inputs)  # get logits, [bs, num_c]

    bs = logits.shape[0]
    for h in hooks:
        h.remove()

    probas = paddle.nn.functional.softmax(logits, axis=1)  # get probabilities.
    preds = paddle.argmax(probas, axis=1)  # get predictions.
    if label is None:
        assert len(preds) == 1
        label = preds.numpy()[:1]  # label is an integer.

    labels = np.concatenate([label] * bs)

    # logits or probas
    labels = labels.reshape((bs, ))
    labels_onehot = paddle.nn.functional.one_hot(paddle.to_tensor(labels), num_classes=probas.shape[1])
    loss = paddle.sum(probas * labels_onehot, axis=1)
    loss.backward()

    gradients = target_feature_map[0].grad  # get gradients of "embedding".
    loss.clear_gradient()

    if isinstance(gradients, paddle.Tensor):
        gradients = gradients.numpy()
    return gradients, labels, target_feature_map[0].numpy(), probas.numpy()
```

paddle同学可以使用下列代码在2.4版本稳定复现错误(需要安装PaddlePaddle/PaddleNLP，PaddlePaddle/TrustAI)：
```shell
pip install paddlenlp
pip install trustai
```
```python
from trustai.demo import DEMO
from trustai.interpretation import IntGradInterpreter
from trustai.interpretation import visualize

demo = DEMO('chnsenticorp')
# init demo model
model = demo.get_model()
tokens, model_inputs = demo("这个宾馆比较陈旧了")

# init interpreter
interpreter = IntGradInterpreter(model)
result = interpreter(model_inputs)


```
错误位置在`${pythonpath}/lib/python3.9/site-packages/trustai/interpretation/token_level/method/integrated_gradients.py#144行`

**Comment by minghaoBD** [comment]
谢谢反馈。

2.4正式版将于近期发布，现在正在做稳定版本的工作。（https://github.com/PaddlePaddle/Paddle/issues/46069 ）建议优先使用2.3.x版本的paddle。2.4的问题我们需要拉相关同学一起看下，有进展了会进行反馈。
